### ISSUE TITLE
Ability to open file from Related section

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#3622]
https://forums.zotero.org/discussion/111240/z7-beta-55-feature-question

We could just add a context menu with "Open PDF in New Tab"/"Open PDF in New Window", but maybe @yexingsha has another idea.

**Comment by yexingsha** [comment]
Context menu can be hard to discover. What if we just add "Open PDF in New Tab" and "Open PDF in New Window" as buttons in the bottom left corner?

And, this entire search dialogue looks... unfinished. I will create a mock for this.

**Comment by dstillman** [comment]
To clarify, this is about opening PDFs from the Related section in the item pane. Right now clicking on an item takes you to the item in the items list. The idea is that it would be nice to be able to go straight to the primary attachment as well.

**Comment by yexingsha** [comment]
> I love the new pop-up search for related items that comes out of clicking the + in the new right panel... I often find myself having to leave what I am reading in the pdf window and go to the Library view in order to do a quick search to see if I have a reference mentioned in the pdf that I am reading... There are probably more slick ways of doing it, but I thought leveraging the Related dialogue search might be pretty straightforward.

So I assumed they were talking about this *search* dialogue that "comes out of clicking the +", which I think adding buttons in the bottom left corner would achieve their desired function:

<img width="621" alt="Screenshot 2024-01-29 at 14 04 44" src="https://github.com/zotero/zotero/assets/57771795/4e57b154-35eb-451d-bedd-3781c099197e">

But for the Related section in the item pane, I agree that adding a context menu that opens the PDF in a new tab/window is a good idea.

**Comment by yexingsha** [comment]
Made the mock, derived from the macOS "save as" dialog which I think suits this dialog perfectly. Since there's a "Cancel" button, we don't really need the window controls here.

<img width="880" alt="related-search-dialog" src="https://github.com/zotero/zotero/assets/57771795/ccc5cf27-91f8-499b-a15b-294e9d9604d3">

**Comment by forensmatt** [comment]
This would be great - Very helpful! 
BTW, I really like how this dialog keeps the search term sticky when changing collections. This is, something that would be very helpful in the main library  panel as well.

**Comment by dstillman** [comment]
> So I assumed they were talking about this search dialogue that "comes out of clicking the +"

Oh, yes, I guess I misread, but then I think it's just a strange, muddled idea that we wouldn't do — if I'm understanding it correctly (which I'm not sure I am), they seem to want to use the related-items dialog as a hacky way to do a library search and open a PDF without actually switching to the library tab? That's not something we would do, nor would we put buttons for that in this dialog — that's not the point of the dialog, and opening a PDF would have to either close the dialog or send it to the back… It just doesn't really make any sense.

There are other ways we could provide a global library-wide search and open PDFs without hijacking an unrelated feature.

The redesigned dialog looks great, though, if we can make a dialog like that — just no "Open PDF" buttons, and no feeds either.

And then this issue — a way to open the primary attachment directly from an entry in the Related item-pane section — isn't what they were asking for, but would be easy enough as a context-menu option.

**Comment by forensmatt** [comment]
> they seem to want to use the related-items dialog as a hacky way to do a library search and open a PDF without actually switching to the library tab?

Yes - that's me :)  I agree it is a little hacky, but the main reason is to be able to maintain focus on the pdf reader tab while looking in my library for a resource referenced in that pdf, i.e.,  do I already have this reference or not? Is it something I wish to make related? If I do have it, then I would like to open the pdf and check the reference, and decide whether to add it as a related item,  or copy the reference to the item note, etc. For my workflow, being able to do that without going to the main library window would be considerably more efficient.  Having many tabs open, it becomes hard to find the tab I was working in after going to the library view.

> a way to open the primary attachment directly from an entry in the Related item-pane section — isn't what they were asking for, but would be easy enough as a context-menu option.

Hence, the context-menu option actually was my original inquiry.  Even for an existing "Related Item" , having a context menu to "Open PDF in New Window"  (or "Open PDF in New Tab") would allow checking the reference without losing focus on the current pdf being read.  Being able to do that from the Add Related search window (without closing the related search window) would then be a great way for checking references as well as checking if I want to include that item in the Related Items list. 

> There are other ways we could provide a way to do a global library-wide search and open PDFs without hijacking an unrelated feature.

Yes, a more global way would be very helpful. Of course, I imagine that could take some time to develop and implement, so I thought the ability to "open the primary attachment directly from an entry in the Related item-pane section...as a context-menu option" would be a great help.
